engine/src/main/java/org/pentaho/di/base/MetaFileLoaderImpl.java

```diff
@@ -0,0 +1,549 @@
+            } else {
+              throw new KettleException(
+                BaseMessages.getString( persistentClass, "JobExecutorMeta.Exception.UnableToLoadJob" ), e );
+            }
+          }
+        }
+        break;
+
+      case REPOSITORY_BY_NAME:
+        String realMetaName = tmpSpace.environmentSubstitute( Const.NVL( metaName, "" ) );
+        String realDirectory = tmpSpace.environmentSubstitute( Const.NVL( directory, "" ) );
+
+        if ( isTransMeta() && space != null ) {
+          // This is a parent transformation and parent variable should work here. A child file name can be
+          // resolved via
+          // parent space.
+          realMetaName = space.environmentSubstitute( realMetaName );
+          realDirectory = space.environmentSubstitute( realDirectory );
+        }
+
+        if ( Utils.isEmpty( realDirectory ) && !Utils.isEmpty( realMetaName ) ) {
+          int index = realMetaName.lastIndexOf( '/' );
+          String transPath = realMetaName;
+          realMetaName = realMetaName.substring( index + 1 );
+          realDirectory = transPath.substring( 0, index );
+        }
+
+        //We will use this key in cache no matter what the final successful path is so that we don't need to hit the
+        //  repo the next time it comes in. (ie: rep.findDirectory )
+        String cacheKey = realDirectory + "/" + realMetaName;
+        theMeta = attemptCacheRead( cacheKey ); //try to get from the cache first
+        if ( theMeta == null ) {
+          if ( rep != null ) {
+            if ( !Utils.isEmpty( realMetaName ) && !Utils.isEmpty( realDirectory ) ) {
+
+              realDirectory = r.normalizeSlashes( realDirectory );
+              RepositoryDirectoryInterface repdir = rep.findDirectory( realDirectory );
+              if ( repdir != null ) {
+                try {
+                  // reads the last revision in the repository...
+                  theMeta = isTransMeta() ? (T) rep.loadTransformation( realMetaName, repdir, null, true, null )
+                    : (T) rep.loadJob( realMetaName, repdir, null, null );
+                  if ( theMeta != null ) {
+                    idContainer[ 0 ] = cacheKey;
+                  }
+                  LogChannel.GENERAL.logDetailed( "Loading " + friendlyMetaType + " from repository",
+                    "Executor " + friendlyMetaType + " [" + realMetaName + "] was loaded from the repository" );
+                } catch ( Exception e ) {
+                  throw new KettleException( "Unable to load " + friendlyMetaType + " [" + realMetaName + "]", e );
+                }
+              }
+            }
+          } else {
+            // rep is null, let's try loading by filename
+            try {
+              theMeta = attemptLoadMeta( cacheKey, rep, metaStore, null, tmpSpace, idContainer );
+            } catch ( KettleException ke ) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)
Similar issue at line numbers 457.